### PR TITLE
refactor: convert confirms to enum

### PR DESCRIPTION
**Pull request by midgleyc**
These are "enum-like", in that they're all alike, and the numbers don't matter, only that they're distinct.

Generate them automatically instead. Most numbers do change, but the specific values don't matter.

This should make adding new ones slightly easier (as now you don't have to care about the string value).

### CODE REVIEW

File: src/net/sourceforge/kolmafia/request/RelayRequest.java [hunk 1]
```diff
@@ -869,15 +876,15 @@ public boolean sendBreakPrismWarning(final String urlString) {
       // If you don't have any coins, nothing to spend
       if ( InventoryManager.getCount( ItemPool.COIN ) > 0 )
       {
-        if ( this.getFormField( CONFIRM_RALPH2 ) == null )
+        if ( this.getFormField( Confirm.CONFIRM_RALPH2 ) == null )
```
Comment (Veracity0) [inline src/net/sourceforge/kolmafia/request/RelayRequest.java:879]: Why is this not Confirm.RALPH2?

File: src/net/sourceforge/kolmafia/request/RelayRequest.java [hunk 2]
```diff
@@ -869,15 +876,15 @@ public boolean sendBreakPrismWarning(final String urlString) {
...
         {
           StringBuilder warning = new StringBuilder();
           warning.append( "When you stop being a plumber, you will lose access to the Mushroom District." );
           warning.append( " Since you have unspent coins, perhaps you would like to visit some shops?" );
           warning.append( " If you are sure you don't want to spend your coins, click the icon on the left. " );
           warning.append( " If you wish to visit the Mushroom District, click on icon on the right." );
           this.sendOptionalWarning(
-            CONFIRM_RALPH2,
+            Confirm.CONFIRM_RALPH2,
```
Comment (Veracity0) [inline src/net/sourceforge/kolmafia/request/RelayRequest.java:887]: Ditto.

File: src/net/sourceforge/kolmafia/request/RelayRequest.java [hunk 3]
```diff
@@ -1240,7 +1251,7 @@ private boolean sendHardcorePVPWarning(final String urlString) {
     }
 
     // Don't remind a second time in a session if you decide not to do it.
-    if (urlString.contains(CONFIRM_HARDCOREPVP)) {
+    if (urlString.contains(Confirm.HARDCOREPVP.toString())) {
```
Comment (Veracity0) [inline src/net/sourceforge/kolmafia/request/RelayRequest.java:1254]: Curious why it is necessary to explicitly call toString here.

File: src/net/sourceforge/kolmafia/request/RelayRequest.java [hunk 4]
```diff
@@ -2602,7 +2613,7 @@ private boolean sendSorceressWarning(final String urlString) {
       return false;
     }
 
-    if (urlString.contains(CONFIRM_SORCERESS)) {
+    if (urlString.contains(Confirm.SORCERESS.toString())) {
```
Comment (Veracity0) [inline src/net/sourceforge/kolmafia/request/RelayRequest.java:2616]: Why explicit toString()?

File: src/net/sourceforge/kolmafia/request/RelayRequest.java [hunk 5]
```diff
@@ -2830,34 +2841,37 @@ private boolean sendSpelunkyWarning(final KoLAdventure adventure) {
...
       return true;
     }
 
     return false;
   }
 
-  public void sendGeneralWarning(final String image, final String message, final String confirm) {
-    this.sendGeneralWarning(image, message, confirm, null, false);
+  public void sendGeneralWarning(final String image, final String message, final Confirm confirm) {
+    this.sendGeneralWarning(image, message, confirm.toString(), null, false);
```
Comment (Veracity0) [inline src/net/sourceforge/kolmafia/request/RelayRequest.java:2861]: OK. These are all turning into calls on the 5-argument version of sendGeneralWarning.
Why not change the third argument of that from ```String confirm``` to ```Confirm confirm```?

**Review by Veracity0 — CHANGES_REQUESTED** [review]

File: src/net/sourceforge/kolmafia/request/RelayRequest.java (hunk 1 above)
Comment (midgleyc) [reply src/net/sourceforge/kolmafia/request/RelayRequest.java:879]: it's commented-out code, so didn't get picked up by the refactor. I'll adjust it.

File: src/net/sourceforge/kolmafia/request/RelayRequest.java (hunk 3 above)
Comment (midgleyc) [reply src/net/sourceforge/kolmafia/request/RelayRequest.java:1254]: `Confirm` doesn't implement `CharSequence`, so it has to be converted to a string. I could implement it, but I thought it was simpler to just convert it.

File: src/net/sourceforge/kolmafia/request/RelayRequest.java (hunk 5 above)
Comment (midgleyc) [reply src/net/sourceforge/kolmafia/request/RelayRequest.java]: I thought the last `sendGeneralWarning` was implemented on `GenericRequest`, but I was wrong. Change made.

**Review by Veracity0 — APPROVED** [review]

**Pull request merged**